### ISSUE TITLE
Check type and value at the same time

### ISSUE THREAD

**Issue by eykd** [kislyuk/ensure#15]
Right now, when checking types with `is_a()`, the return value is `None` for a scalar value, and an `IterableInspector` for an iterable value. In neither case can I then go on to check the value of the subject.

For a scalar:

```
In [9]: ensure(9).is_an(int).which.equals(9)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-9-c205f4e9cc21> in <module>()
----> 1 ensure(9).is_an(int).which.equals(9)

AttributeError: 'NoneType' object has no attribute 'which'
```

For an iterable:

```
In [7]: ensure('foo').is_a(str).which.equals('foo')
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-7-48f991a6e079> in <module>()
----> 1 ensure('foo').is_a(str).which.equals('foo')

AttributeError: 'IterableInspector' object has no attribute 'which'
```

The examples are trivial, but I find myself wanting to check the type and value frequently as a part of a chain. I'm used to being able to do this with a fluent assertion interface like [Chai](http://chaijs.com/api/bdd/), and I'd love to see ensure reach that level of fluency.

A good start would be to never return None, returning the wrapped subject instead, and to add `which` and `whose_value` properties on all the inspectors which take you back to the wrapped subject again.

**Comment by kislyuk** [comment]
Thanks for the suggestion. How does that look?

**Comment by eykd** [comment]
Definitely better. What do you think of making each method return something, even `self`? That way you can stack assertions, e.g.:

```
(ensure(9)
 .is_greater_than(1)
 .is_less_than(10))
```

**Comment by kislyuk** [comment]
I'd like to keep the assertion language grammatically correct wherever feasible, so I'd rather return another proxy inspector that can access the subject via `.and`: `ensure(9).is_greater_than(1).and.is_less_than(10)`

**Comment by eykd** [comment]
With `and` being a keyword, will that work? `and_` would, but that's a bit ugly. I'll be the first to applaud grammatically correct APIs--that's part of the joy of a fluent interface. But don't forget that ["practicality beats purity"](https://www.python.org/dev/peps/pep-0020/).

That being said, the design decision lies with you. Thanks for releasing this library--I find it useful on a weekly basis. I look forward to seeing these updates hit PyPI!

**Comment by kislyuk** [comment]
Yes, you're right. I'm thinking of using "also" instead of "and".

**Comment by kislyuk** [comment]
Released in v0.4.0, please test.
